var/spack/repos/builtin/packages/py-urllib3/package.py

```diff
@@ -23,22 +25,33 @@ class PyUrllib3(PythonPackage):
     version("1.20", sha256="97ef2b6e2878d84c0126b9f4e608e37a951ca7848e4855a7f7f4437d5c34a72f")
     version("1.14", sha256="dd4fb13a4ce50b18338c7e4d665b21fd38632c5d4b1d9f1a1379276bd3c08d37")
 
-    variant("socks", default=False, description="SOCKS and HTTP proxy support")
+    variant("brotli", default=False, when="@1.25:", description="Add Brotli support")
     variant("secure", default=False, description="Add SSL/TLS support")
-    variant("brotli", default=False, description="Add Brotli support")
-
-    depends_on("python@2.7:2.8,3.4:", when="@:1.25", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.5:", when="@1.26.6", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.6:3", when="@1.26.12:", type=("build", "run"))
-
-    depends_on("py-setuptools", type="build")
-
-    depends_on("py-pyopenssl@0.14:", when="+secure")
-    depends_on("py-cryptography@1.3.4:", when="+secure")
-    depends_on("py-idna@2:", when="+secure")
-    depends_on("py-certifi", when="+secure")
-    depends_on("py-urllib3-secure-extra", when="+secure @1.26.12:")
-
-    depends_on("py-pysocks@1.5.6,1.5.8:1", when="+socks")
-
-    depends_on("py-brotlipy@0.6:", when="+brotli")
+    variant("socks", default=False, when="@1.15:", description="SOCKS and HTTP proxy support")
+
+    depends_on("py-hatchling@1.6:1", when="@2:", type="build")
+
+    with when("+brotli"):
+        depends_on("py-brotli@1.0.9:", when="@2", type=("build", "run"))
+
+        # Historical dependencies
+        depends_on("py-brotli@1.0.9:", when="@1.26 platform=linux", type=("build", "run"))
+        depends_on("py-brotli@1.0.9:", when="@1.26 platform=darwin", type=("build", "run"))
+        depends_on("py-brotli@1.0.9:", when="@1.26 platform=cray", type=("build", "run"))
+        depends_on("py-brotlipy@0.6:", when="@1.26 platform=windows", type=("build", "run"))
```

Comment:
I think this is only when Python < 3. Since Python > 3, the dep will always be py-brotli on all OSes.